### PR TITLE
Reuse api-model classes in systemtests

### PR DESCRIPTION
**Pull request by kornys**

### CODE REVIEW

**Comment by kornys** [comment]
@enmasse-ci run tests

**Comment by kornys** [comment]
@enmasse-ci run tests

**Comment by kornys** [comment]
@enmasse-ci run tests

**Comment by kornys** [comment]
@enmasse-ci run tests

**Comment by kornys** [comment]
@enmasse-ci run tests testcase=**.infra.InfraTest

File: api-server/src/main/java/io/enmasse/api/v1/http/HttpSchemaService.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 import javax.ws.rs.core.Response;
 
 /**
- * API for serving address model schema.
+ * API for serving address utils schema.
```
Comment (lulf) [inline api-server/src/main/java/io/enmasse/api/v1/http/HttpSchemaService.java]: Probably unintentional change

File: systemtests/scripts/run_tests.sh [hunk 2]
```diff
@@ -6,4 +6,4 @@ if [[ -n "$TESTCASE" ]]; then
     EXTRA_ARGS="-Dtest=$TESTCASE"
 fi
 
-mvn test -P${PROFILE} -Djava.net.preferIPv4Stack=true -Djansi.force=true -Dstyle.color=always ${EXTRA_ARGS}
+cd .. && mvn test -pl systemtests -P${PROFILE} -am -Djava.net.preferIPv4Stack=true -Djansi.force=true -DfailIfNoTests=false -Dstyle.color=always ${EXTRA_ARGS}
```
Comment (lulf) [inline systemtests/scripts/run_tests.sh]: This mutates current dir. The expectation is that one is still in the same dir afterwards, so cd back into system tests

**Comment by kornys** [comment]
@enmasse-ci run tests

**Comment by kornys** [comment]
@lulf all POJO were rewritten to using api-model, tests fails dues to problem with creating addressSpace based on mixed addressspaceplan 
```
    io.fabric8.kubernetes.client.KubernetesClientException: An error has occurred.
            at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:64) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.KubernetesClientException.launderThrowable(KubernetesClientException.java:53) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:238) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:191) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:384) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:344) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:328) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.dsl.base.BaseOperation.listRequestHelper(BaseOperation.java:165) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:607) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.dsl.base.BaseOperation.list(BaseOperation.java:67) ~[address-space-controller.jar:na]
            at io.enmasse.k8s.api.KubeAddressPlanApi.list(KubeAddressPlanApi.java:34) ~[address-space-controller.jar:na]
            at io.enmasse.k8s.api.KubeAddressPlanApi.list(KubeAddressPlanApi.java:20) ~[address-space-controller.jar:na]
            at io.enmasse.k8s.api.cache.Reflector.resync(Reflector.java:78) ~[address-space-controller.jar:na]
            at io.enmasse.k8s.api.cache.Reflector.run(Reflector.java:53) ~[address-space-controller.jar:na]
            at io.enmasse.k8s.api.cache.Controller.lambda$start$0(Controller.java:22) ~[address-space-controller.jar:na]
            at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
    Caused by: com.fasterxml.jackson.databind.JsonMappingException: N/A
     at [Source: (BufferedInputStream); line: 1, column: 21801] (through reference chain: io.enmasse.admin.model.v1.AddressPlanList["items"]->java.util.ArrayList[17]->io.enmasse.admin.model.v1.AddressPlan["resources"])
            at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:278) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.SettableBeanProperty._throwAsIOE(SettableBeanProperty.java:611) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.impl.SetterlessProperty.deserializeAndSet(SetterlessProperty.java:122) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:286) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:245) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.std.CollectionDeserializer.deserialize(CollectionDeserializer.java:27) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.impl.MethodProperty.deserializeAndSet(MethodProperty.java:127) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:288) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:151) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4013) ~[address-space-controller.jar:na]
            at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3077) ~[address-space-controller.jar:na]
            at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:236) ~[address-space-controller.jar:na]
    ... 13 common frames omitted

```

**Comment by lulf** [comment]
I know the issue now. It requires some more fixing, i will do this in the unit test PR, but it would conflict with #2415  so i will merge that first.

**Comment by lulf** [comment]
@kornys if you rebase this on master, you should get correct serialization of the plans

**Comment by kornys** [comment]
@lulf nice, thank you!

**Comment by kornys** [comment]
@enmasse-ci run tests

**Pull request merged**